src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMBuilder.java

```diff
@@ -206,70 +188,130 @@ public BitbucketGitSCMBuilder withCredentials(String credentialsId, BitbucketRep
      */
     @NonNull
     public BitbucketGitSCMBuilder withBitbucketRemote() {
-        SCMHead h = head();
-        String repoOwner;
-        String repository;
-        BitbucketApi bitbucket = scmSource().buildBitbucketClient();
-        if (h instanceof PullRequestSCMHead && bitbucket instanceof BitbucketCloudApiClient) {
-            // TODO fix once Bitbucket Cloud has a fix for https://bitbucket.org/site/master/issues/5814
-            repoOwner = ((PullRequestSCMHead) h).getRepoOwner();
-            repository = ((PullRequestSCMHead) h).getRepository();
+        SCMHead head = head();
+        withoutRefSpecs();
+        String headName = head.getName();
+        if (head instanceof PullRequestSCMHead) {
+            withPullRequestRemote((PullRequestSCMHead) head, headName);
+        } else if (head instanceof TagSCMHead) {
+            withTagRemote(headName);
         } else {
-            // head instanceof BranchSCMHead
-            repoOwner = scmSource.getRepoOwner();
-            repository = scmSource.getRepository();
+            withBranchRemote(headName);
         }
+        return this;
+    }
 
-        String cloneLink = null;
-        for (BitbucketHref link : cloneLinks()) {
-            if (protocol.getType().equals(link.getName())) {
-                cloneLink = link.getHref();
-                break;
+    private void withPullRequestRemote(PullRequestSCMHead head, String headName) {
+        String scmSourceRepoOwner = scmSource.getRepoOwner();
+        String scmSourceRepository = scmSource.getRepository();
+        String pullRequestRepoOwner = head.getRepoOwner();
+        String pullRequestRepository = head.getRepository();
+        boolean prFromTargetRepository = pullRequestRepoOwner.equals(scmSourceRepoOwner)
```

Comment:
@andrey-fomin I've run into an issue with this.

When configuring a job with a Bitbucket branch source, if you enter the "Owner" the input is case-insensitive.
The "Repository Name" field below it will be populated correctly.

But then when you actually build a same-origin PR, `prFromTargetRepository` will return `false`, leading to very hard to diagnose issues.

In my case I was severely confused when my refs showed as:
```
"+refs/pull-requests/" + pullId + "/from:refs/remotes/@{remote}/" + headName
```
instead of the expected:
```
"+refs/heads/" + branchName + ":refs/remotes/@{remote}/" + branchName
```

All because I had entered the "Owner" as lower-case (which the Jenkins job config happily accepted), whereas in Bitbucket it is upper-case.

I think this line should be:
```
boolean prFromTargetRepository = pullRequestRepoOwner.equalsIgnoreCase(scmSourceRepoOwner)
    && pullRequestRepository.equalsIgnoreCase(scmSourceRepository);
```